[5/N] Add update in method

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11534
* #11533
* __->__ #11463
* #11462
* #11391
* #11390
* #11389

Expose the update API in method. This API will take BackendOptionMap as input and call backend_interface->update under the hood.

Differential Revision: [D76149467](https://our.internmc.facebook.com/intern/diff/D76149467/)

cccclai committed:
[5/N] Add update in method

Expose the update API in method. This API will take BackendOptionMap as input and call backend_interface->update under the hood.

Differential Revision: [D76149467](https://our.internmc.facebook.com/intern/diff/D76149467/)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add update in method"


Expose the update API in method. This API will take BackendOptionMap as input and call backend_interface->update under the hood.

Differential Revision: [D76149467](https://our.internmc.facebook.com/intern/diff/D76149467/)

[ghstack-poisoned]

cccclai committed:
Update on "[5/N] Add update in method"


Expose the update API in method. This API will take BackendOptionMap as input and call backend_interface->update under the hood.

Differential Revision: [D76149467](https://our.internmc.facebook.com/intern/diff/D76149467/)

[ghstack-poisoned]

kimishpatel commented on runtime/executor/method.cpp:
```diff
@@ -1512,6 +1513,26 @@ Error Method::experimental_step() {
+    const char* backend_name = entry.backend_name;
+    auto backend_options = entry.options;
+
+    auto backend_class = get_backend_class(backend_name);
```
does backend class give you a singleton instance of backend interface? If so why do we need to pipe it up via Method class. Just make something like `executorch::backend::set_options(BackendOptions&)` (I dont think BackendOptions constructor exist? but probably would be good to add)

digantdesai replied:
I agree. Allowing it to pass via Method or Module hints that it is private to a method which in fact it isn't.

cccclai committed:
Update on "[5/N] Add update in method"


Expose the update API in method. This API will take BackendOptionMap as input and call backend_interface->update under the hood.

Differential Revision: [D76149467](https://our.internmc.facebook.com/intern/diff/D76149467/)

[ghstack-poisoned]
